src/reconcile/rec_visibility.c

```diff
@@ -179,22 +193,24 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins,
 		 * globally visible, need to check the update state as well.
 		 */
 		if (F_ISSET(r, WT_REC_EVICT)) {
-			if (upd->prepare_state == WT_PREPARE_LOCKED ||
-			    upd->prepare_state == WT_PREPARE_INPROGRESS) {
-				list_prepared = true;
-				continue;
-			}
 			if (F_ISSET(r, WT_REC_VISIBLE_ALL) ?
 			    WT_TXNID_LE(r->last_running, txnid) :
 			    !__txn_visible_id(session, txnid)) {
 				r->update_uncommitted = list_uncommitted = true;
 				continue;
 			}
+			if (upd->prepare_state == WT_PREPARE_LOCKED ||
+			    upd->prepare_state == WT_PREPARE_INPROGRESS) {
+				list_prepared = true;
+				if (upd->start_ts > max_ts)
+					max_ts = upd->start_ts;
+				continue;
+			}
```

Comment:
I'd like to talk this part through.  My reasoning for this reordering is that we should consistently check that the transaction ID is committed, then look at `upd->prepare_state`, then look at `start_ts` and/or `durable_ts`.